app/ssttpaffordability/AffordabilityForm.scala

```diff
@@ -18,25 +18,29 @@ package ssttpaffordability
 
 import play.api.data.Forms.{text, _}
 import play.api.data.{Form, FormError, Mapping}
+import ssttpaffordability.model.IncomeCategory
+import ssttpaffordability.model.IncomeCategory.{Benefits, MonthlyIncome, OtherIncome}
 import ssttpaffordability.model.forms.helper.FormErrorWithFieldMessageOverrides
-
 import scala.util.Try
 
 object AffordabilityForm {
 
   val incomeForm: Form[IncomeInput] = Form(
     mapping(
-      "monthlyIncome" -> validateIncome(text),
-      "benefits" -> validateIncome(text),
-      "otherIncome" -> validateIncome(text)
+      "monthlyIncome" -> validateIncome(text)(MonthlyIncome),
+      "benefits" -> validateIncome(text)(Benefits),
+      "otherIncome" -> validateIncome(text)(OtherIncome)
     )(IncomeInput.apply)(IncomeInput.unapply)
   )
 
-  private def validateIncome(mappingStr: Mapping[String]) = {
-    mappingStr.verifying("ssttp.affordability.your-monthly-income.error.non-numerals", { i: String =>
+  private def validateIncome(mappingStr: Mapping[String])(incomeCategory: IncomeCategory) = {
```

Comment:
I'm just being picky, but do we need 2 parameter lists? could it be this and fix the 3 calls to this methods.  
`private def validateIncome(mappingStr: Mapping[String], incomeCategory: IncomeCategory)`